File: src/main/java/fr/formation/stravacat/model/Animal.java
package fr.formation.model;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonView;

import fr.formation.api.Views;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;

@Entity
@Table(name = "animal")
public class Animal {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "anim_id", nullable = false)
    @JsonView(Views.Common.class)
    private int id;

    @JsonView(Views.Animal.class)
    @Column(name = "anim_pseudo", nullable = false)
    @NotBlank
    private String pseudo;

    @JsonView(Views.AnimalDetail.class)
    @Column(name = "anim_email", nullable = false)
    @NotBlank
    private String email;
    
    @Column(name = "anim_password", nullable = false)
    private String password;
    
    @JsonView(Views.Animal.class)
    @Column(name = "anim_age", nullable = false)
    @Positive
    private int age;
    
    @JsonView(Views.Animal.class)
    @Column(name = "anim_espece", nullable = false)
    @NotBlank
    private String espece;
    
    @JsonView(Views.AnimalDetail.class)
    @ManyToOne
    @JoinColumn(name = "anim_ville_id", nullable = false)
    @Valid
    private Ville ville;

    @JsonView(Views.AnimalDetail.class)
    @OneToMany(mappedBy = "animal")
    private List<Parcours> parcours;

    @JsonView(Views.AnimalDetail.class)
    @OneToMany(mappedBy = "animal")
    private List<Actualite> actualites;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getPseudo() {
        return pseudo;
    }

    public void setPseudo(String pseudo) {
        this.pseudo = pseudo;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public String getEspece() {
        return espece;
    }

    public void setEspece(String espece) {
        this.espece = espece;
    }

    public Ville getVille() {
        return ville;
    }

    public void setVille(Ville ville) {
        this.ville = ville;
    }

    public List<Parcours> getParcours() {
        return parcours;
    }

    public void setParcours(List<Parcours> parcours) {
        this.parcours = parcours;
    }

    public List<Actualite> getActualites() {
        return actualites;
    }

    public void setActualites(List<Actualite> actualites) {
        this.actualites = actualites;
    }
}

File: angular/src/app/models/animal.ts
import { Actualite } from "./actualite";
import { Parcours } from "./parcours";
import { Ville } from "./ville";

export class Animal {
    public get actualites(): Actualite[] {
        return this._actualites;
    }
    public set actualites(value: Actualite[]) {
        this._actualites = value;
    }
    public get parcours(): Parcours[] {
        return this._parcours;
    }
    public set parcours(value: Parcours[]) {
        this._parcours = value;
    }
    public get ville(): Ville {
        return this._ville;
    }
    public set ville(value: Ville) {
        this._ville = value;
    }
    public get espece(): string {
        return this._espece;
    }
    public set espece(value: string) {
        this._espece = value;
    }
    public get age(): number {
        return this._age;
    }
    public set age(value: number) {
        this._age = value;
    }
    public get isAdmin(): boolean {
        return this._isAdmin;
    }
    public set isAdmin(value: boolean) {
        this._isAdmin = value;
    }
    public get password(): string {
        return this._password;
    }
    public set password(value: string) {
        this._password = value;
    }
    public get email(): string {
        return this._email;
    }
    public set email(value: string) {
        this._email = value;
    }
    public get pseudo(): string {
        return this._pseudo;
    }
    public set pseudo(value: string) {
        this._pseudo = value;
    }
    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }

    constructor(
        private _id: number,
        private _pseudo: string,
        private _email: string,
        private _password: string,
        private _isAdmin: boolean,
        private _age: number,
        private _espece: string,
        private _ville: Ville,
        private _parcours: Parcours[],
        private _actualites: Actualite[]
        ) { }
}
